Migration Guide for Lock v11 & Auth0.js v9

aaguiarz opened the pull request:
Trying new approach for Migration Guides

https://auth0-docs-content-pr-5406.herokuapp.com/docs/libraries/auth0js/v9/migration-guide
https://auth0-docs-content-pr-5406.herokuapp.com/docs/libraries/lock/v11/migration-guide

aaguiarz committed:
Added new Migration Guide for Lock v11

aaguiarz committed:
Updated guide

aaguiarz committed:
Added Auth0.js migration guide, and fixed link in Lock one

aaguiarz committed:
Several improvements

aaguiarz committed:
Improved auth0.js guide

aaguiarz committed:
Refactored Auth0.js 9 guide include steps per version

aaguiarz committed:
Minor improvement

aaguiarz committed:
Fixed spaces between links

aaguiarz committed:
Improved layout

aaguiarz committed:
Formatting

aaguiarz committed:
Shortened description

aaguiarz committed:
Added links to migration guides

aaguiarz committed:
- Added lock migration guides and various adjustments

aaguiarz committed:
- Added angular/react guides

aaguiarz committed:
Fixed issue with included file

aaguiarz committed:
Fixing included file

aaguiarz committed:
Fixed links

aaguiarz committed:
Fixing wording

aaguiarz committed:
Improved wording

aaguiarz committed:
Minor fixes

aaguiarz committed:
Fixed broken links

luisrudge commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,49 @@
+
+- Centralized Login, using the `.authorize()` method, where the user is redirected from your website to an Auth0's Hosted Login Page.
+
+- Embedded Login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.
```
remove `client.loginWithCredentials()`. We have it just to be backwards compatible. new apps should use just `.login`

aaguiarz replied:
The idea is to communicate "if I'm using some of those methods then I need to worry about migrating", so I think we need to keep it

luisrudge commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,49 @@
+| Embedded Login | Required |
+| Customized Hosted Page | Not Supported |
+
+If you are using Centralized Login, you don't need to make any changes in your code to use auth0.js v9, you can just update it to the latest version.
```
isn't the latest version v9? since v9 is not supported in the HLP, this message is a bit confusing

aaguiarz replied:
I clarified it, let me know if it's more clear

luisrudge commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,49 @@
+
+[Migrating from Auth0.js v8](/articles/libraries/auth0js/v9/migration-v8-v9)
+
+[Migrating to Auth0.js v9 in Angular.js Applications](/articles/libraries/auth0js/v9/migration-angularjs)
```
there's two angular links

aaguiarz committed:
Fixes in getSSOData doc

aaguiarz committed:
Clarified scenario regarding updating auth0.js when you use .authorize()

aaguiarz committed:
Fixed casing of auth0.js

aaguiarz committed:
Fix in getSSOData doc

aaguiarz committed:
Fixes in getSSOData()

aaguiarz committed:
Minor fixes

aaguiarz committed:
Added guides for different versions of Angular

aaguiarz committed:
Added missing file

aaguiarz committed:
Fixing links

aaguiarz committed:
Fixed typo

aaguiarz committed:
Fixed typos

aaguiarz committed:
Improved default values article
Fixed typo

aaguiarz committed:
Assorted fixes

aaguiarz committed:
Minor rewording

aaguiarz committed:
Improved the list of migration guides
Added explanation to why you need to configure web origins

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+---
+# Migrating to Auth0.js v9
+
+Auth0.js v9 has been improved to work better in embedded login scenarios. It operates with enhanced security and removes dependencies that have been deprecated as per Auth0's roadmap. In some cases, these security enhancements may impact application behavior when upgrading from an earlier versions of Auth0.js. 
```
Link "embedded login scenarios" to `/guides/login/centralized-vs-embedded`?

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+
+Auth0.js can be used to implement authentication in different ways:
+
+- Centralized Login, using the `.authorize()` method, where the user is redirected from your website to an Auth0's Hosted Login Page.
```
At the risk of overstating things, I'd say 

> In your app, to trigger Centralized Login using the `.authorize()` method, where the user [...]

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+
+- Centralized Login, using the `.authorize()` method, where the user is redirected from your website to an Auth0's Hosted Login Page.
+
+- Embedded Login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.
```
and here 

> In your app, to implement Embedded Login using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.

aaguiarz replied:
What do you mean, @nicosabena ?

nicosabena replied:
Sorry, I meant replacing

> Embedded Login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.

with

> In your app, to implement Embedded Login using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.

aaguiarz replied:
OK sorry, it's done.. thanks!

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+| Centralized Login | Recommended |
+| Embedded Login | Required |
+| Customized Hosted Page | Not Supported |
+
```
Again, I would replace with:

| In your app, to trigger Centralized Login | Recommended |
| In your app, to implement Embedded Login | Required |
| In a customized Hosted Login Page | Not Supported, keep using Auth0.js v8 and/or Lock v10 |

mpaktiti committed:
Fix header level

mpaktiti committed:
Fix header level, make table headers bold

mpaktiti committed:
Remove toc, the article is too short

mpaktiti committed:
Fix typo

mpaktiti committed:
Add short intro sentence

mpaktiti committed:
Add intro text

mpaktiti committed:
Add intro sentence

mpaktiti committed:
Replace Lock.js with Lock

mpaktiti committed:
Replace Lock.js

mpaktiti committed:
Replace Lock.js, fix broken link, remove toc

mpaktiti committed:
Replace Lock.js, remove toc, fix broken link

mpaktiti committed:
Remove .js from Lock

jmangelo commented on articles/libraries/_includes/_configure_embedded_login.md:
```diff
@@ -0,0 +1,10 @@
+
+Lock v11 and Auth0.js v9 use cross-origin calls inside hidden iframes to perform authentication. To make sure this can be done securely, the Auth0 sever needs to know the domains where you will be hosting your applications.
+
+Add the domain to the **Allowed Web Origins** field. You can find this field in the [Client Settings](${manage_url}/#/clients/${account.clientId}/settings).
```
I would be a little more specific then just _domain_ as that setting requires the user to set a web origin as defined by the spec (scheme, host and port when using a non-default port for the scheme). Ideally, this should have two examples, one with default ports and another with non-default ports. Something like:

* if you're running your application at `https://example.com/portal` then you need to add the web origin `https://example.com`
* if you're running your application at `https://example.com:5503/portal` then you need to add the web origin `https://example.com:5503`

An explicit warning that the path component should not be included even if it's just `/` might be useful as I already seen people adding `https://example.com/` when it should not contain the last `/`.

mpaktiti committed:
Replace Lock.js, remove whitespace

mpaktiti committed:
Replace Lock.js

mpaktiti committed:
Remove Lock.js references

mpaktiti committed:
Remove .js from Lock, fix broken links, remove toc

mpaktiti committed:
Remove .js from Lock

mpaktiti committed:
Remove .js from Lock

mpaktiti committed:
Remove .js from Lock

mpaktiti committed:
Fix typo

mpaktiti committed:
Copyedits, add links to tokens docs

mpaktiti committed:
Format changes, add link

mpaktiti committed:
Add link to scopes

mpaktiti committed:
Add fullstops

mpaktiti committed:
Add fullstops

mpaktiti committed:
Add fullstop

mpaktiti committed:
Add fullstop

mpaktiti committed:
Add intro text

mpaktiti committed:
Remove duplicate header

mpaktiti commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+
+Auth0.js v9 has been improved to work better in embedded login scenarios. It operates with enhanced security and removes dependencies that have been deprecated as per Auth0's roadmap. In some cases, these security enhancements may impact application behavior when upgrading from an earlier versions of Auth0.js. 
+
+Auth0.js can be used to implement authentication in different ways:
```
This sentence looks a bit out of context with the previous paragraph. I would add a header like `## Should I migrate to v9?`

mpaktiti commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,55 @@
+| **Scenario** | **Migration to v9** | 
+| --- | --- | 
+| Centralized Login | Recommended |
+| Embedded Login | Required |
```
The capitalization is not consistent. `Centralized Login` is the same in all docs I saw, but embedded can be either `Embedded Login` or `embedded login`. I would have them consistent (personally I prefer `centralized login` and `embedded login`, with links to the `/guides/login/centralized-vs-embedded`)

mpaktiti committed:
Fix broken links, remove toc

mpaktiti committed:
Add intro text

mpaktiti committed:
Add intro text

mpaktiti committed:
Add intro text

mpaktiti committed:
Format changes

mpaktiti committed:
Add links to scopes doc, format changes

mpaktiti reviewed (approved):
LGTM. I did some copyediting and left a couple of comments.

nicosabena commented:
In the individual migration guides we should add a common header at the top that links to the main document (`/docs/libraries/auth0js/v9/migration-guide` / `docs/libraries/lock/v11/migration-guide`).

Something like:

> Auth0.js v9 has been improved to work better in embedded login scenarios, with improved security and removed dependencies on legacy features. Find out if you should upgrade or not by reading [Migrating to Auth0.js v9](/docs/libraries/auth0js/v9/migration-guide).

(Similar for Lock)

nicosabena commented on articles/libraries/lock/v11/migration-guide.md:
```diff
@@ -0,0 +1,34 @@
+Lock 11 is designed for **embedded login** scenarios. It operates with enhanced security and removes dependencies that have been deprecated as per Auth0's roadmap. In some cases, these security enhancements may impact application behavior when upgrading from an earlier version of Lock. 
+
+We recommend that instead of using Lock embedded in your application, you use **Centralized Login**, as it is the [most secure, powerful and flexible approach for authentication](/guides/login/centralized-vs-embedded).
+
```
We should add a warning similar to what's found on /docs/libraries/auth0js/v9/migration-guide, mentioning that if you are using Lock to customize the Hosted Login Page, then you need to stick with v10. We have two scenarios now:

* If using Lock embedded in your application -> you must upgrade
* If using Lock in the customized Hosted Login Page (i.e. for centralized login) -> keep using Lock v10

nicosabena commented on articles/libraries/_includes/_get_lock_latest_version.md:
````diff
@@ -0,0 +1,23 @@
+
+```html
+<script src="${lock_url}"></script>
+```
````
Careful, this still points to Lock v10.20

aaguiarz replied:
We'll update the variable to Lock11 when we ship

jeffreylees replied:
Yes, this variable will be updated as soon as we have a release version to list

aaguiarz committed:
Incorporated feedback from Nico/Maria

aaguiarz committed:
Em

aaguiarz committed:
Remove abbreviation

aaguiarz committed:
Minor fixes

aaguiarz committed:
Incorporated feedback from DSE

aaguiarz committed:
Removed extra whitespace

thameera commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,60 @@
+
+- In your application, to implement embedded login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.
+
+- In the [Hosted Login Page](/hosted-pages/login), where you can use the same methods as in embedded login but from inside a customized Auth0's Hosted Login Page. Most customers don't customize Auth0 Hosted Login Page with auth0.js, so your probably don't need to worry about this scenario.
```
> so your probably

thameera commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,60 @@
+If your application is configured to use centralized login, and your web application uses `.authorize()`, you can update auth0.js to v9 in your web application, without requiring any code changes.
+
+::: warning
+If you are not using centralized login in your applications, we recommend you to start using it, it as [most secure, powerful and flexible approach for authentication](/guides/login/centralized-vs-embedded). 
```
> it, it as

nicosabena replied:
At the risk of being repetitive, I'd change the above paragraph to this, to clarify the move from embedded login to centralized login:

> If you are using login embedded in your applications, we recommend moving to a centralized login approach, as it is the [most secure, powerful and flexible approach for authentication](/guides/login/centralized-vs-embedded).

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,60 @@
+
+- In the [Hosted Login Page](/hosted-pages/login), where you can use the same methods as in embedded login but from inside a customized Auth0's Hosted Login Page. Most customers don't customize Auth0 Hosted Login Page with auth0.js, so your probably don't need to worry about this scenario.
+
+Depending on how you are using auth0.js, you have different options:
```
As they aren't really "options', I'd say:

> Migration to v9 will depend on how you are using auth0.js:

nicosabena commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,60 @@
+| **Scenario** | **Migration to v9** | 
+| --- | --- | 
+| In your application, to trigger centralized login | Recommended |
+| In your application, to implement embedded login | Required |
```
How about

> Required (read below)

instead of just

> Required

to make sure the centralized login argument below is more emphasized?

aaguiarz committed:
Added guide from v9 SPA

nicosabena commented on articles/libraries/lock/v11/migration-guide.md:
```diff
@@ -0,0 +1,51 @@
+---
+# Migrating to Lock v11
+
+[Lock 11](/libraries/lock) is designed for [embedded login** scenarios](/guides/login/centralized-vs-embedded). It operates with enhanced security and removes dependencies that have been deprecated as per Auth0's roadmap. In some cases, these security enhancements may impact application behavior when upgrading from an earlier version of Lock. 
```
embedded login**

nicosabena commented on articles/libraries/lock/v11/migration-guide.md:
```diff
@@ -0,0 +1,51 @@
+
+| **Scenario** | **Migration to v9** | 
+| --- | --- | 
+| In your application, to implement embedded login | Required |
```
Similarly to the Auth0.js migration docs, I'd emphasize just a tad more the alternative of moving to centralized login instead of upgrading Lock, by replacing

> Required

with

> Required (read below)

coupled with the other edit to the warning panel

aaguiarz committed:
Layout fixes and broken links

nicosabena commented on articles/libraries/lock/v11/migration-guide.md:
```diff
@@ -0,0 +1,51 @@
+| In a customized Hosted Login Page | Not Supported, keep using Lock 10 and/or Auth0.js |
+
+::: warning
+If you are not using centralized login in your applications, we recommend you to start using it, it as [most secure, powerful and flexible approach for authentication](/guides/login/centralized-vs-embedded). 
```
> If you are using Lock to implement login embedded in your applications, we recommend moving to a centralized login approach, as it is the [most secure, powerful and flexible approach for authentication](/guides/login/centralized-vs-embedded).

aaguiarz committed:
Reordered guides

aaguiarz committed:
Reworded recommendation to use access tokens

aaguiarz committed:
Extracted non oidc migration content

aaguiarz committed:
Rephrased warning

aaguiarz committed:
Fixed links

aaguiarz committed:
Incorporated feedback

aaguiarz committed:
Renamed OIDC Conformant migration doc

aaguiarz committed:
Changed non-oidc conformant to legacy

jhiner commented on articles/libraries/lock/v11/migration-legacy-flows.md:
```diff
@@ -0,0 +1,118 @@
+
+You can check the 'Calling an API' section of our [SPA Quickstarts](/quickstart/backend) for more information on how to call APIs from SPAs. You will also need to migrate your backend API implementation to use access_tokens. You can look at our [API Quickstarts](/quickstart/backend) for instructions on how to do it.
+
+## Using OIDC-Conformant User Profiles
```
To me the section could just be `User Profiles`

jhiner commented on articles/libraries/lock/v11/migration-legacy-flows.md:
````diff
@@ -0,0 +1,118 @@
+    );
+```
+
+If you specify an audience, then the OIDC flow will be triggered and the user profile data returned by Auth0 in `id_tokens` or from `/userinfo` will be OIDC-compliant. If your application is using any non-standard claim from the user profile, it will break. Read [below](#using-oidc-conformant-user-profiles) for more information on how to deal with this issue.
````
> OIDC-compliant
Would stick with OIDC-conformant

aaguiarz committed:
- Consolidated all Lock 9 migration scenarions in a single doc
- Incorporated feedback from Justin

mpaktiti committed:
Fix URL, add toc, some copyedits

mpaktiti committed:
Fix header levels

mpaktiti committed:
Format changes

luisrudge committed:
adding code samples to a0js v7-v9 migration

aaguiarz committed:
Added guides for Angular v9 and fixed minor things in auth0.js v7-v9

mpaktiti committed:
Add variables for clientId and domain

aaguiarz committed:
Changed auth0.js 6->9 guide

mpaktiti commented on articles/libraries/auth0js/v9/migration-v7-v9.md:
```diff
@@ -8,182 +8,170 @@ toc: true
-  domain: "${account.namespace}",
-  clientID: "${account.clientId}",
-  responseType: "token"
+  domain:'${account.namespace}'),,
```
Hey @aaguiarz the `),` characters (in the `domain:'${account.namespace}'),,`) look like a typo

mpaktiti commented on articles/libraries/lock/v11/migration-angularjs-v9.md:
```diff
@@ -4,7 +4,192 @@ title: Migrating AngularJS Applications to from Lock v9 to Lock v11
+
+## 4. Updating Authentication Service
+
+If you are using an authentication service, you will need to make a few minor changes there as well. Lock 11 provides events so you can execute functionality when the user is authenticated, when there is an authentication error, etc. You can read more at the [Lock 11 documentation](sdk).
```
@aaguiarz Is `sdk` meant to be `/libraries/lock/v10`?

aaguiarz replied:
Yes, just fixed it

mpaktiti committed:
Add ui element for note

aaguiarz committed:
Polished Angular guides

aaguiarz committed:
Merge branch 'aa-migration-guides' of https://github.com/auth0/docs into aa-migration-guides

aaguiarz committed:
Added link to Lock 11 docs

aaguiarz committed:
Added doc on how to call Mgmt API with an access_token

aaguiarz committed:
Minor adjustments

aaguiarz committed:
Adding guides for Lock v8

aaguiarz committed:
Adjusted wording

jhiner commented on articles/libraries/lock/v11/migration-legacy-flows.md:
```diff
@@ -0,0 +1,171 @@
+
+In the legacy flows, the [Management API](/api/management/v2) supported authentication with an `id_token`. This approach has been deprecated, and now you need to call it with an `access_token`. 
+
+To get an get an `access_token`, you need to ask Auth0 for one using the `https://'${account.namespace}/api/v2/` audience. Auth0 does not currently support specifying two audiences when authenticating, so you will need to still use your application's API audience when initializing Lock or auth0.js. Once the user is authenticated, you can use `checkSession` to retrieve a Management API `access_tok ...
```
`https://'${account.namespace}/api/v2/`

^ is there an extra single quote ' in there?

mpaktiti replied:
Yes there was (fixed), good catch 👍

aaguiarz committed:
Minor fixes

aaguiarz committed:
Updated vars.yml to have Lock v11 Auth0.js v9 urls

aaguiarz committed:
Minor fixes

mpaktiti committed:
Fix typo

aaguiarz committed:
Added library flag in _configure_embedded_login

aaguiarz committed:
Merge branch 'aa-migration-guides' of https://github.com/auth0/docs into aa-migration-guides

aaguiarz committed:
Update _configure_embedded_login.md

jeffreylees commented on articles/libraries/_includes/_configure_embedded_login.md:
```diff
@@ -0,0 +1,13 @@
+### Configure Auth0 for Embedded Login
+
+<% if (library == 'lock') { %>
```
Is there a reason to separate these out with a conditional when they say practically the same thing? Could it be something like this?

>Cross-origin calls inside hidden iframes will be used to perform authentication. To make sure this can be done securely, Auth0 needs to know the domains where you will be hosting your applications.

jeffreylees committed:
Updating wording for clarity

jeffreylees commented on articles/libraries/auth0js/v9/migration-guide.md:
```diff
@@ -0,0 +1,60 @@
+
+- In your application, to implement embedded login, using the `.login()`, `popup.loginWithCredentials()`, `client.loginWithCredentials()` methods, where the login dialog is displayed in the application's website.
+
+- In the [Hosted Login Page](/hosted-pages/login), where you can use the same methods as in embedded login but from inside a customized Auth0's Hosted Login Page. Most customers don't customize Auth0 Hosted Login Page with auth0.js, so you probably don't need to worry about this scenario.
```
>Most customers... probably don't need to worry about this scenario

Seems to be a statement that's not necessary here - if this option wasn't worth saying, we wouldn't present it as an option, and since it is here, we probably should just leave it at that and not add the caveat that "most customers" don't need it.

jeffreylees committed:
Syntax updates

jeffreylees committed:
Updates to wording and a typo correction

aaguiarz committed:
Fixing condition

aaguiarz committed:
Removed if condition

aaguiarz committed:
Added library parameter and refactored migration guide introduction

aaguiarz committed:
Updated support matrix to indicate that Lock 11 is the only supported library.

Updated getSSOdata doc to note that it does not work with response_type = 'code'.

aaguiarz committed:
Reworded recommendation for checkSession usage

aaguiarz committed:
Merge branch 'master' into aa-migration-guides

aaguiarz committed:
Added old versions of files
